### PR TITLE
Storage metrics implementation

### PR DESCRIPTION
**Pull request by g4s8**
#403 - added `Metrics` implementation based on `Storage`, moved logs metrics publisher.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#404](https://github.com/artipie/artipie/issues/404) is already in scope

File: src/main/java/com/artipie/metrics/asto/StorageMetrics.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                                .string(StandardCharsets.UTF_8)
+                                .thenApply(Long::valueOf)
+                        );
+                    } else {
+                        res = CompletableFuture.completedFuture(0L);
+                    }
+                    return res;
+                }
+            ).thenCompose(val -> this.storage.save(this.key, content(val + amount)))
+                .handle(StorageDatabase::handle);
```
Comment (olegmoz) [inline src/main/java/com/artipie/metrics/asto/StorageMetrics.java:116]: @g4s8 this approach to updating values will lead to race conditions right away (the method is called at least once for each request, that's very often) and very high performance overhead (one memory update is 3 storage ops). I believe that buffering ans synchronizing writes is mandatory here from the start.

What do you think about collecting data to `InMemoryMetrics` and publishing it periodically to `Storage` approximately the same way we publish it to log? That way we have buffering and synchronized writes.

**Comment by g4s8** [comment]
@olegmoz thanks, should be fixed now

**Comment by 0crat** [comment]
Job [#404](https://github.com/artipie/artipie/issues/404) is not in the agenda of @olegmoz[/z](https://www.0crat.com/u/olegmoz), can't set title

**Comment by 0crat** [comment]
Job audit: performer @olegmoz[/z](https://www.0crat.com/u/olegmoz)[/z](https://www.0crat.com/u/olegmoz) didn't make CR comments

**Pull request merged**